src/parse.c

```diff
@@ -1488,6 +1571,17 @@ const mapping_entry_handler vlan_def_handlers[] = {
     {NULL}
 };
 
+const mapping_entry_handler tunnel_def_handlers[] = {
+    COMMON_LINK_HANDLERS,
+    {"mode", YAML_SCALAR_NODE, handle_tunnel_mode},
+    {"parent", YAML_SCALAR_NODE, handle_netdef_id_ref, NULL, netdef_offset(tunnel.parent)},
+    {"local", YAML_SCALAR_NODE, handle_netdef_ip4, NULL, netdef_offset(tunnel.local_ip)},
+    {"remote", YAML_SCALAR_NODE, handle_netdef_ip4, NULL, netdef_offset(tunnel.remote_ip)},
+    {"input-key", YAML_SCALAR_NODE, handle_tunnel_key, NULL, netdef_offset(tunnel.input_key)},
+    {"output-key", YAML_SCALAR_NODE, handle_tunnel_key, NULL, netdef_offset(tunnel.output_key)},
```

Comment:
networkd supports 'key' as shorthand for setting input-key and output-key to the same value; and the draft suggested we would support all three names.  Did you decide against this?

One possibility that comes to mind is to allow 'key' to take either a scalar (key: $mykey) or a dict (key: input: $inputkey output: $outputkey).

Reply:
WIP. Still does need to add key. I like the idea of it being a scalar or dict, if we can convince libyaml to figure it out (I think it's possible).